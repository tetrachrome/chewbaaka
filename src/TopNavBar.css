/**
 * TopNavBar.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 02, 2020
 */

.ui.menu.topNavBarMenu {
  background-color: rgba(229, 239, 249, 0.65);
}

nav.ui.container .ui.menu .item {
  padding-bottom: 0;
}

nav.ui.container .ui.menu .item {
  font-size: 16px;
  padding-left: 0px;
  padding-right: 0px;
  padding-bottom: 14px;
  margin-left: 16px;
  margin-right: 16px;
  border-radius: 0;
}
