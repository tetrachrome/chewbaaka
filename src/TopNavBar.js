/**
 * TopNavBar.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 02, 2020
 */

import React from 'react';

import { Container, Image, Menu } from 'semantic-ui-react'

import logo from './logo.svg'

class TopNavBar extends React.Component {
  render() {
    return (
      <Menu fixed="top" className="topNavBarMenu">
        <Container>
          <Menu.Item as="a" header href="https://cheetah.org">
            <Image
              size="small"
              src={logo}
              style={{ marginRight: "1.5em" }}
            />
          </Menu.Item>
        <Menu.Item as="a" href="https://www.iucnredlist.org/species/219/50649567" target="_blank">IUCN Status</Menu.Item>
        </Container>
      </Menu>
    )
  }
}

export default TopNavBar;
