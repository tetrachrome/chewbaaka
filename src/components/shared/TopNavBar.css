/**
 * TopNavBar.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 28, 2020
 */

.ui.menu.TopNavBarMenu {
  background-color: rgba(229, 239, 249, 0.65);
}
