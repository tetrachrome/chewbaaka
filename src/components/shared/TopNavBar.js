/**
 * TopNavBar.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 13, 2020
 */

import React from 'react';

import { Container, Image, Menu } from 'semantic-ui-react'

import logo from './assets/cheetah-conservation-fund-30-logo.svg'

import './TopNavBar.css';

const __TEST__ = ( process.env.NODE_ENV === "test" );

export default class TopNavBar extends React.Component {

  render() {
    /**
     * Set `fixed` prop of `Menu` component to "top"
     * in test mode to avoid warning about invalid prop.
     *
     * TODO: Figure out a better solution for this ...
     */
    const menuFixedStatus = ( this.props.fixedOnTop || __TEST__ ) ? "top": "";

    return (
      <Menu fixed={menuFixedStatus} className="topNavBarMenu" data-testid="TopNavBarComponentTestId">
        <Container>
          <Menu.Item as="a" header href="https://cheetah.org" data-testid="mainNavBarItem">
            <Image
              size="small"
              src={logo}
              style={{ marginRight: "1.5em" }}
            />
          </Menu.Item>
          <Menu.Item as="a" href="https://www.iucnredlist.org/species/219/50649567" target="_blank">IUCN Status</Menu.Item>
          <Menu.Item as="a" href="https://wildnet.org/wildlife-programs/cheetah-namibia/" target="_blank">Wildlife Conservation Network</Menu.Item>
          <Menu.Item as="a" href="https://cheetah.org/about/what-we-do/conservation/" target="_blank">Conservation</Menu.Item>
        </Container>
      </Menu>
    )
  }
}
