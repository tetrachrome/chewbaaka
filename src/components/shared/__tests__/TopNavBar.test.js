/**
 * TopNavBar.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 08, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import TopNavBar from '../TopNavBar';

test('renders top nav bar', () => {
  render(<TopNavBar />);

  // Tests first/main nav bar item.
  {
    const navBarItemMainItem = screen.getByTestId('mainNavBarItem');
    expect(navBarItemMainItem).toBeInTheDocument();

    // Expect main nav bar item to point to CCF site.
    expect(navBarItemMainItem.href).toBe("https://cheetah.org/");
  }

  // Tests second nav bar item.
  {
    const navBarItemIUCNStatus = screen.getByText(/IUCN status/i);
    expect(navBarItemIUCNStatus).toBeInTheDocument();

    // Expect this nav bar item point to IUCN Red List page.
    expect(navBarItemIUCNStatus.href).toBe("https://www.iucnredlist.org/species/219/50649567");
  }

  // Tests third nav bar item.
  {
    const navBarItemWCN = screen.getByText(/Wildlife Conservation Network/i);
    expect(navBarItemWCN).toBeInTheDocument();

    // Expect this nav bar item point to WCN site.
    expect(navBarItemWCN.href).toBe("https://wildnet.org/wildlife-programs/cheetah-namibia/");
  }
});

test('TopNavBar component snapshot', () => {
  const tree = renderer
    .create(
      <TopNavBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
