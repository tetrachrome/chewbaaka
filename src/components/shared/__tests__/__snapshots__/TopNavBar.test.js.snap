// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNavBar component snapshot 1`] = `
<div
  className="ui top fixed topNavBarMenu menu"
  data-testid="TopNavBarComponentTestId"
>
  <div
    className="ui container"
  >
    <a
      className="header item"
      data-testid="mainNavBarItem"
      href="https://cheetah.org"
      onClick={[Function]}
    >
      <img
        className="ui small image"
        src="cheetah-conservation-fund-30-logo.svg"
        style={
          Object {
            "marginRight": "1.5em",
          }
        }
      />
    </a>
    <a
      className="item"
      href="https://www.iucnredlist.org/species/219/50649567"
      onClick={[Function]}
      target="_blank"
    >
      IUCN Status
    </a>
    <a
      className="item"
      href="https://wildnet.org/wildlife-programs/cheetah-namibia/"
      onClick={[Function]}
      target="_blank"
    >
      Conservation
    </a>
    <a
      className="item"
      href="https://cheetah.org/get-involved/ways-to-give/"
      onClick={[Function]}
      target="_blank"
    >
      Get Involved
    </a>
  </div>
</div>
`;
